dao-macros/src/instance/mod.rs

```diff
@@ -0,0 +1,98 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{spanned::Spanned, DataStruct, DeriveInput, Ident};
+
+pub fn generate_code(input: DeriveInput) -> TokenStream {
+    match generate_or_err(input) {
+        Ok(tokens) => tokens,
+        Err(err) => err.to_compile_error(),
+    }
+}
+
+fn generate_or_err(input: DeriveInput) -> Result<TokenStream, syn::Error> {
+    let span = input.span();
+    match input.data {
+        syn::Data::Struct(data_struct) => parse_data(input.ident, data_struct),
+        syn::Data::Enum(_) => Err(syn::Error::new(span, "Cannot instantiate an enum")),
+        syn::Data::Union(_) => Err(syn::Error::new(span, "Cannot instantiate a union")),
+    }
+}
+
+fn parse_data(struct_ident: Ident, data_struct: DataStruct) -> Result<TokenStream, syn::Error> {
+    let fields: Vec<TokenStream> = data_struct.fields.into_iter().map(|field| {
+        let ident = field.ident.unwrap();
+        quote! {
+            #ident: casper_dao_utils::instance::Instanced::instance(&format!("{}:{}", stringify!(#ident), namespace))
+        }
+    }).collect();
+
+    Ok(quote! {
+        impl casper_dao_utils::instance::Instanced for #struct_ident {
+
+            fn instance(namespace: &str) -> Self {
+                Self {
+                    #( #fields, )*
+                }
+            }
+        }
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use quote::quote;
+    use syn::punctuated::Punctuated;
+
+    use super::parse_data;
+
+    #[test]
+    fn parsing_struct_data_works() {
```

Comment:
This test uses full paths to structs (like `syn::Token`), while everywhere else you use just structs. So it's inconsistent.